builder/processor/src/main/java/io/helidon/builder/processor/TypeHandler.java

```diff
@@ -0,0 +1,391 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.builder.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import io.helidon.common.types.TypeName;
+
+import static io.helidon.builder.processor.Types.CHAR_ARRAY_TYPE;
+import static io.helidon.builder.processor.Types.DURATION_TYPE;
+import static io.helidon.builder.processor.Types.LIST_TYPE_NAME;
+import static io.helidon.builder.processor.Types.MAP_TYPE_NAME;
+import static io.helidon.builder.processor.Types.OPTIONAL_TYPE;
+import static io.helidon.builder.processor.Types.STRING_TYPE;
+
+class TypeHandler {
+    static final String UNCONFIGURED = "io.helidon.config.metadata.ConfiguredOption.UNCONFIGURED";
+    private final String name;
+    private final String getterName;
+    private final String setterName;
+    private final TypeName declaredType;
+
+    TypeHandler(String name, String getterName, String setterName, TypeName declaredType) {
+        this.name = name;
+        this.getterName = getterName;
+        this.setterName = setterName;
+        this.declaredType = declaredType;
+    }
+
+    static TypeHandler create(String name, String getterName, String setterName, TypeName returnType, boolean sameGeneric) {
+        if (OPTIONAL_TYPE.equals(returnType)) {
+            return new TypeHandlerOptional(name, getterName, setterName, returnType);
+        }
+        if (Types.SET_TYPE_NAME.equals(returnType)) {
+            return new TypeHandlerSet(name, getterName, setterName, returnType);
+        }
+
+        if (Types.LIST_TYPE_NAME.equals(returnType)) {
+            return new TypeHandlerList(name, getterName, setterName, returnType);
+        }
+        if (Types.MAP_TYPE_NAME.equals(returnType)) {
+            return new TypeHandlerMap(name, getterName, setterName, returnType, sameGeneric);
+        }
+
+        return new TypeHandler(name, getterName, setterName, returnType);
+    }
+
+    static String setterModifier(PrototypeProperty.ConfiguredOption configured) {
+        if (configured.builderMethod()) {
+            return "public ";
+        }
+        return "";
+    }
+
+    static TypeName toWildcard(TypeName typeName) {
+        if (typeName.wildcard()) {
+            return typeName;
+        }
+        return TypeName.builder(typeName).wildcard(true).build();
+    }
+
+    protected static String collectionImplType(TypeName typeName) {
+        TypeName genericTypeName = typeName.genericTypeName();
+        if (genericTypeName.equals(MAP_TYPE_NAME)) {
+            return "java.util.LinkedHashMap";
```

Comment:
these should be static definitions somewhere. Same for below ArrayList, and LinkedHashSet.

Reply:
pushed